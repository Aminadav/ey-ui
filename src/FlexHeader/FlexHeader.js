import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Grid from '@material-ui/core/Grid';
import withStyles from '@material-ui/core/styles/withStyles';

const FlexHeader = ({ children, alignRight, borderBottom, marginTop, classes }) => {
  const classNames = classnames(classes.root, {
    [classes.alignRight]: alignRight,
    [classes.borderBottom]: borderBottom,
    [classes.marginTop]: marginTop,
  });

  return (
    <Grid data-testid="flex-header" container className={classNames}>
      {children}
    </Grid>
  );
};

FlexHeader.propTypes = {
  alignRight: PropTypes.bool,
  borderBottom: PropTypes.bool,
  marginTop: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.node]).isRequired,
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    marginTop: PropTypes.string.isRequired,
    borderBottom: PropTypes.string.isRequired,
  }).isRequired,
};

FlexHeader.defaultProps = {
  alignRight: false,
  borderBottom: false,
  marginTop: false,
};

const styles = () => ({
  root: {
    marginTop: 0,
    marginBottom: 0,
  },
  marginTop: {
    margin: '5rem 0 0 0',
  },
  borderBottom: {
    borderBottom: '1px solid silver',
  },
  alignRight: {
    justifyContent: 'flex-end',
  },
});

export default withStyles(styles)(FlexHeader);
