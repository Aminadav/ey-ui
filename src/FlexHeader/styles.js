import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(() => ({
  root: {
    marginTop: 0,
    marginBottom: 0,
  },
  marginTop: {
    margin: '5rem 0 0 0',
  },
  borderBottom: {
    borderBottom: '1px solid silver',
  },
  alignRight: {
    justifyContent: 'flex-end',
  },
}));

export default useStyles;
