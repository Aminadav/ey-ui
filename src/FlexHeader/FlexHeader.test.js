import React from 'react';
import { render } from '@testing-library/react';
import FlexHeader from './FlexHeader';

describe('FlexHeader', () => {
  it('matches the snapshot with no props', () => {
    const { container } = render(
      <FlexHeader>
        <div>Test</div>
      </FlexHeader>,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot with marginTop prop', () => {
    const { container } = render(
      <FlexHeader marginTop>
        <div>Test</div>
      </FlexHeader>,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot with borderBottom prop', () => {
    const { container } = render(
      <FlexHeader borderBottom>
        <div>Test</div>
      </FlexHeader>,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot with both marginTop and borderBottom prop', () => {
    const { container } = render(
      <FlexHeader borderBottom marginTop>
        <div data-testid="flex-header-content">Test</div>
      </FlexHeader>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should allow for text input as children', () => {
    const { container } = render(<FlexHeader>TestContent</FlexHeader>);
    expect(container).toHaveTextContent('TestContent');
  });

  it('should allow for array input as children', () => {
    const { container } = render(<FlexHeader>{['ArrayElement1', 'ArrayElement2']}</FlexHeader>);
    expect(container).toHaveTextContent('ArrayElement1');
    expect(container).toHaveTextContent('ArrayElement2');
  });

  it('should have expected styles when alignRight prop is set to true', () => {
    const { getByTestId } = render(
      <FlexHeader alignRight>
        <div>TestContents</div>
      </FlexHeader>,
    );
    const grid = getByTestId('flex-header');
    expect(grid).toHaveStyle('justify-content: flex-end');
  });
});
