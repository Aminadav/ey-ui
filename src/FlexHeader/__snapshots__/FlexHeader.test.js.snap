// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexHeader matches the snapshot with borderBottom prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-215 makeStyles-borderBottom-217 MuiGrid-container"
    data-testid="flex-header"
  >
    <div>
      Test
    </div>
  </div>
</div>
`;

exports[`FlexHeader matches the snapshot with both marginTop and borderBottom prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-322 makeStyles-borderBottom-324 makeStyles-marginTop-323 MuiGrid-container"
    data-testid="flex-header"
  >
    <div
      data-testid="flex-header-content"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`FlexHeader matches the snapshot with marginTop prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-108 makeStyles-marginTop-109 MuiGrid-container"
    data-testid="flex-header"
  >
    <div>
      Test
    </div>
  </div>
</div>
`;

exports[`FlexHeader matches the snapshot with no props 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
    data-testid="flex-header"
  >
    <div>
      Test
    </div>
  </div>
</div>
`;
